cpp/example_code/transfer-manager/transferOnStream.cpp

```diff
@@ -86,35 +88,53 @@ int main(int argc, char** argv)
 
         auto uploadHandle = transfer_manager->UploadFile(LOCAL_FILE, BUCKET, KEY, "text/plain", Aws::Map<Aws::String, Aws::String>());
         uploadHandle->WaitUntilFinished();
-        std::cout << "File upload finished!" << std::endl;
-
-        // verify upload expected length of data
-        assert(uploadHandle->GetBytesTotalSize() == uploadHandle->GetBytesTransferred());
-        g_file_size = uploadHandle->GetBytesTotalSize();
-
-        // This buffer is what we used to initialize streambuf and is in memory
-        Aws::Utils::Array<unsigned char> buffer(BUFFER_SIZE);
-        auto downloadHandle = transfer_manager->DownloadFile(BUCKET,
-                KEY, 
+        bool success = uploadHandle->GetStatus() == Transfer::TransferStatus::COMPLETED; 
+      
+        if (!success)
+        {
+            auto err = uploadHandle->GetLastError();           
+            std::cout << "File upload failed:  "<< err.GetMessage() << std::endl;
+        }
+        else
+        {
+            std::cout << "File upload finished." << std::endl;
+
+            // Verify upload expected length of data
+            assert(uploadHandle->GetBytesTotalSize() == uploadHandle->GetBytesTransferred());
+            g_file_size = uploadHandle->GetBytesTotalSize();
+
+            // This buffer is what we used to initialize streambuf and is in memory
+            Aws::Utils::Array<unsigned char> buffer(BUFFER_SIZE);
+            auto downloadHandle = transfer_manager->DownloadFile(BUCKET,
+                KEY,
                 [&]() { //create stream lambda fn
```

Comment:
Create stream(ing?) Lambda function. Use "AWS Lambda" if this is the first use in this file.

Reply:
No, i'm pretty sure they don't mean AWS Lambda here. I think they mean a lambda syntax coding construct.  Regardless, you're right about the comment, whoever wrote it, its pretty poor :(  I went with 
//Define a lambda expression for the callback method parameter to stream back the data.